src/parser/ast.ml

```diff
@@ -13,21 +23,33 @@ type inline_element =
   | `Reference of
     reference_kind * string with_location * inline_element with_location list
   | `Link of string * inline_element with_location list ]
+(** Inline elements are equivalent to what would be found in a [span] in HTML.
+    Mostly these are straightforward. The [`Reference] constructor takes a triple
+    whose second element is the reference itself, and the third the replacement
+    text. Similarly the [`Link] constructor has the link itself as first parameter
+    and the second is the replacement text. *)
 
 type nestable_block_element =
   [ `Paragraph of inline_element with_location list
-  | `Code_block of string
+  | `Code_block of string option * string
   | `Verbatim of string
   | `Modules of string with_location list
   | `List of
     [ `Unordered | `Ordered ]
     * [ `Light | `Heavy ]
     * nestable_block_element with_location list list ]
+(** Some block elements may be nested within lists or tags, but not all.
+    The [`List] constructor has a parameter of type [\[`Light | `Heavy\]].
+    This corresponds to the syntactic constructor used (see the
+    {{:https://ocaml.org/releases/4.12/htmlman/ocamldoc.html#sss:ocamldoc-list}manual}).
+    *)
 
 type internal_tag =
   [ `Canonical of string with_location | `Inline | `Open | `Closed ]
+(** Internal tags are used to exercise fine control over the output of odoc. They
+    are never rendered in the output *)
 
-type external_tag =
+type ocamldoc_tag =
```

Comment:
I think this reference to `ocamldoc` isn't necessary, we can call this `tag` instead.